src/api/app/controllers/source_project_package_meta_controller.rb

```diff
@@ -1,7 +1,26 @@
 class SourceProjectPackageMetaController < SourceController
   before_action :require_package_name, only: [:show, :update]
+  before_action :set_rdata, only: [:update]
+  before_action :validate_project_name, only: [:update]
+  before_action :validate_package_name, only: [:update]
   validate_action update: { request: :package, response: :status }
 
+  def set_rdata
+    @rdata = Xmlhash.parse(request.raw_post)
+  end
+
+  def validate_package_name
+    err_message = 'package name in xml data does not match resource path component'
+    err_code = 'package_name_mismatch'
+    render_error status: 400, errorcode: err_code, message: err_message if @rdata['name'] && @rdata['name'] != @package_name
+  end
+
+  def validate_project_name
+    err_message = 'project name in xml data does not match resource path component'
+    err_code = 'project_name_mismatch'
+    render_error status: 400, errorcode: err_code, message: err_message if @rdata['project'] && @rdata['project'] != @project_name
```

Comment:
why `if @rdata['project']`?

Reply:
this goes back to ceff3ced0c and should really die

Reply:
@coolo  what you mean, it should die? Could we remove it completely?

Reply:
as Ana wrote. the != is good enough - no need to check for nil

Reply:
I would go with @marcus-h and keep the nil check there.